src/Build/Definition/ProjectProperty.cs

```diff
@@ -94,7 +96,18 @@ public string EvaluatedValue
         string IProperty.EvaluatedValueEscaped
         {
             [DebuggerStepThrough]
-            get => EvaluatedValueEscapedInternal;
+            get
+            {
+                if (this is EnvironmentDerivedProjectProperty environmentProperty && environmentProperty.loggingContext?.IsValid == true && !environmentProperty._loggedEnvProperty)
```

Comment:
I'd maybe do `environmentProperty.loggingContext is { IsValid: true } loggingContext` and then just use `loggingContext` to avoid repeating `environmentProperty.loggingContext`

https://en.wikipedia.org/wiki/Law_of_Demeter